tests/test_nexus_to_json.py

```diff
@@ -330,3 +332,124 @@ def test_GIVEN_string_list_WHEN_getting_data_and_type_THEN_returns_correct_dtype
+    pv_name = "pv1"
+    topic = "localhost:9092/someTopic"
+    writer_module = "f142"
+    streams = {
+        "stream1": {"writer_module": writer_module, "source": pv_name, "topic": topic}
+    }
+
+    dummy_file = io.StringIO()
+    generate_forwarder_command(dummy_file, streams, "ca")
+
+    streams_ = literal_eval(dummy_file.getvalue())["streams"]
+    assert len(streams_) == 1
+    assert streams_[0]["channel"] == pv_name
+    assert streams_[0]["converter"]["topic"] == topic
+    assert streams_[0]["converter"]["schema"] == writer_module
+
+
+def test_GIVEN_stream_with_f142_command_and_non_forwarder_modules_THEN_only_f142_is_contained():
+    pv_name = "pv1"
+    topic = "localhost:9092/someTopic"
+    writer_module = "f142"
+    streams = {
+        "stream1": {"writer_module": writer_module, "source": pv_name, "topic": topic},
+        "stream2": {"writer_module": "ev42", "source": "source1", "topic": "topic1"},
+    }
+
+    dummy_file = io.StringIO()
+    generate_forwarder_command(dummy_file, streams, "ca")
+
+    streams_ = literal_eval(dummy_file.getvalue())["streams"]
+    assert len(streams_) == 1
+    assert streams_[0]["channel"] == pv_name
+    assert streams_[0]["converter"]["topic"] == topic
+    assert streams_[0]["converter"]["schema"] == writer_module
+
+
+def test_GIVEN_stream_with_tdc_command_WHEN_generating_forwarder_command_THEN_output_contains_pv():
+    pv_name = "tdcpv1"
+    topic = "localhost:9092/someOtherTopic"
+    writer_module = "TdcTime"
+    streams = {
+        "stream1": {"writer_module": writer_module, "source": pv_name, "topic": topic}
+    }
+
+    dummy_file = io.StringIO()
+    generate_forwarder_command(dummy_file, streams, "pva")
+
+    streams_ = literal_eval(dummy_file.getvalue())["streams"]
+    assert len(streams_) == 1
+    assert streams_[0]["channel"] == pv_name
+    assert streams_[0]["converter"]["topic"] == topic
+    assert streams_[0]["converter"]["schema"] == writer_module
+
+
+def test_GIVEN_stream_with_one_pv_with_two_topics_WHEN_generating_forwarder_command_THEN_contains_one_converter_with_list():
+    pv_name = "testPV"
+
+    topic1 = "topic1"
+    topic2 = "topic2"
+
+    writer_module = "f142"
+
+    streams = {
+        "stream1": {"writer_module": writer_module, "source": pv_name, "topic": topic1},
+        "stream2": {"writer_module": writer_module, "source": pv_name, "topic": topic2},
+    }
+
+    dummy_file = io.StringIO()
+    generate_forwarder_command(dummy_file, streams, "ca")
+
+    streams_ = literal_eval(dummy_file.getvalue())["streams"]
+
+    assert len(streams_) == 1
+
+    assert isinstance(streams_[0]["converter"], list)
+    assert streams_[0]["channel"] == pv_name
+    assert streams_[0]["converter"][0]["topic"] == topic1
+    assert streams_[0]["converter"][1]["topic"] == topic2
+
+
+def test_GIVEN_stream_with_pv_forwarding_to_three_topics_WHEN_generating_forwarder_command_THEN_stream_is_added_to_converters():
```

Comment:
What does this test that the previous test doesn't?

Reply:
Covers this branch: 
https://github.com/ess-dmsc/nexus-constructor/pull/462/files#diff-3b148aa9d555ba73e4949e927cd13dd9R55-R56